🪟 🧹 Implement ColumnSortButton component

YatsukBogdan1 opened the pull request:
## What
Close #17062 and #17491

## How
There was already a codebase prepared for this, but it wasn't organized into a single component, so this PR adds a new `SortableHeaderComponent` that is located inside `ui/Table` folder

<img width="1331" alt="Screenshot 2022-10-17 at 18 25 13" src="https://user-images.githubusercontent.com/25852567/196219521-465e1bd8-0b6f-499c-abe7-db703d0e97ae.png">

YatsukBogdan1 committed:
Implement ColumnSortButton component

YatsukBogdan1 committed:
Merge branch 'master' into byatsuk/feature-create-sort-column-button

edmundito commented on airbyte-webapp/src/components/ColumnSortButton/index.tsx:
```diff
@@ -0,0 +1,24 @@
+  lowToLarge: boolean;
+}
+
+export const ColumnSortButton: React.FC<ColumnSortButtonProps> = ({
```
This is not just the button but also contains the text of the heading, the name should be `SortableTableHeader`. This should also be part of the Table export, so it's probably best if it goes into `components/ui/Table`

edmundito replied:
If we're using the children as suggested below, in order to make this compatible with React 18:

```suggestion
export const ColumnSortButton: React.FC<React.PropsWithChildren<ColumnSortButtonProps>> = ({
```

edmundito commented on airbyte-webapp/src/components/ColumnSortButton/index.tsx:
```diff
@@ -0,0 +1,24 @@
+  lowToLarge,
+}) => (
+  <button className={styles.sortButton} onClick={onClick}>
+    <FormattedMessage id={formattedMessageId} />
```
The preference is not to pass the messageId but instead just pass the content, so one suggestion is to make this `{children}` and let the user of this component add `<FormattedMessage id="...." />` as the child.

edmundito commented on airbyte-webapp/src/components/ColumnSortButton/index.tsx:
```diff
@@ -0,0 +1,24 @@
+}) => (
+  <button className={styles.sortButton} onClick={onClick}>
+    <FormattedMessage id={formattedMessageId} />
+    <SortIcon wasActive={wasActive} lowToLarge={lowToLarge} />
```
Is it necessary to have a sort icon component file? It looks like it's very simple and it could be implemented here.

edmundito reviewed (commented):
Tested changes locally, and looks good. Left some comments about the location and some improvements to the file.

YatsukBogdan1 committed:
Merge branch 'master' into byatsuk/feature-create-sort-column-button

YatsukBogdan1 committed:
Updates component name; Moves component to ui/Table folder; Refactors formattedMessageId property into using render content as children directly; Removes minor SortIcon component

terencecho committed:
Skip psql stop in acceptance test for gke (#18023)

ryankfu committed:
Checks for iterator hasNext element (#18041)

* Checks for iterator hasNext element

* Fix linter with newline

edmundito commented on airbyte-webapp/src/App.tsx:
```diff
@@ -62,6 +62,7 @@ const Services: React.FC<React.PropsWithChildren<unknown>> = ({ children }) => (
 );
 
 const App: React.FC = () => {
+  console.log("started");
```
```suggestion
```

gosusnp committed:
Add Message Migration to Destination Connection Checks (#17954)

* Add Message Migration to Destination Connection Checks

* Fix test setup

xpuska513 committed:
Update helm release workflow (#18048)

* Update workflow

* Update trigger rules

* fix: Update release workflow with abillity to add tags

* Update workflow

evantahler committed:
Remove unused `airbyte-cli` (#18009)

edmundito commented on airbyte-webapp/src/components/ui/Table/SortableTableHeader/index.tsx:
```diff
@@ -0,0 +1,23 @@
+
+interface SortableTableHeaderProps {
+  onClick: () => void;
+  wasActive: boolean;
+  lowToLarge: boolean;
```
I noticed that these names came from the old component but they could be more clear:

`wasActive` => `isActive`
`lowToLarge` => `isAscending`

dizel852 replied:
Agree, need to rename this couple

girarda committed:
🐛  [low-code] $options shouldn't overwrite values that are already defined (#18060)

* fix

* Add missing test

* remove prints

* extract to method

* rename

* Add missing test

* rename

* bump

xpuska513 committed:
Update helm chart comments (#18072)

xpuska513 committed:
Update helm charts (#18073)

* add test

* fix chart.yaml

suhomud committed:
16250 Destination Redis: Add SSH support (#17951)

* 16250 Destination Redis: Add SSH support

* 16250 Resolve port issue

* 11679 Bump version

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

github-actions[bot] committed:
Bump helm chart version reference to 0.40.20 (#18074)

* Bump helm chart version reference to 0.40.20

* remove binary

Co-authored-by: xpuska513 <xpuska513@users.noreply.github.com>
Co-authored-by: Kyryl Skobylko <xpuska513@gmail.com>

sfc-gh-pkommini committed:
Helm Chart: Create service annotations for airbyte-server (#17932)

* Support annotations for airbyte-server as well, update version and update docs.

* Fix auto-indent.

Co-authored-by: Kyryl Skobylko <xpuska513@gmail.com>

benmoriceau committed:
Bmoric/remove dep server worker (#17894)

* test [ci skip]

* Autogenerated files

* Add missing annotation

* Remove unused json2Schema block from worker

* Move tess

* Missing deps and format

* Fix test build

* TMP

* Add missing dependencies

* PR comments

* Tmp

* [ci skip] Tmp

* Fix acceptance test and add the seed dependency

* Fix build

* For diff

* tmp

* Build pass

* make the worker to be  on the platform only

* fix setting.yaml

* Fix pmd

* Fix Cron

* Add chart

* Fix cron

* Fix server build.gradle

* Fix jar conflict

* PR comments

* Add cron micronaut environemnt

Amruta-Ranade committed:
Updated connector catalog page (#18076)

benmoriceau committed:
Move the port forward outside of the main docker-compose (#17864)

octavia-squidington-iii committed:
Bump Airbyte version from 0.40.14 to 0.40.15 (#17970)

Co-authored-by: benmoriceau <benmoriceau@users.noreply.github.com>

artem1205 committed:
🎉 Source Shopify: Add metafield streams (#17962)

* 🎉 Source Shopify: Add metafield streams

* Source Shopify: fix unittest

* Source Shopify: docs update

* Source Shopify: fix backward compatibility test

* Source Shopify: fix schemas

* Source Shopify: fix state filter

* Source Shopify: refactor & optimize

* Source Shopify: fix test privileges

* Source Shopify: fix stream filter

* Source Shopify: fix streams

* Source Shopify: update abnormal state

* Source Shopify: fix abnormal state streams

* Source Shopify: fix streams

* updated methods, formated code

* Source Shopify: typo fix

* auto-bump connector version

Co-authored-by: Oleksandr Bazarnov <oleksandr.bazarnov@globallogic.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

brianjlai committed:
fix check for streams that do not use a stream slicer (#18080)

* fix check for streams that do not use a stream slicer

* increment version and changelog before publish

mfsiega-airbyte committed:
tolerate database nulls in webhook operation configs (#18084)

mfsiega-airbyte committed:
Implement webhook operation in the sync workflow (#18022)

Implements the webhook operation as part of the sync workflow.

- Introduces the new activity implementation
- Updates the various interfaces that pass input to get the relevant configs to the sync workflow
- Hooks the new activity into the sync workflow
- Passes the webhook configs along into the sync workflow job

github-actions[bot] committed:
Bump helm chart version reference to 0.40.22 (#18077)

alexander-marquardt committed:
Added new "filters" python file, along with a "hash" filter. This can… (#18000)

* Added new "filters" python file, along with a "hash" filter. This can be extended to include other custom filters in the future.

* Added additional comments

* Moved usage of the hash_obj inside the conditional that confirms it exists

* Moved the hash function call inside a condition to ensure that it exists

* Fixed the application of the salt , so that it does not modify the hash unless it is actually passed in.

* Added unit tests to validate new jinja hash functionality

* Updated unit test to pass numeric value as a float instead of string

* Removed unreferenced import to pytest

* Updated version

* format

* format

* format

* format

* format

Co-authored-by: Alexandre Girard <alexandre@airbyte.io>

github-actions[bot] committed:
Bump helm chart version reference to 0.40.24 (#18081)

* Bump helm chart version reference to 0.40.24

* Update .gitignore

Co-authored-by: benmoriceau <benmoriceau@users.noreply.github.com>
Co-authored-by: Kyryl Skobylko <xpuska513@gmail.com>

davydov-d committed:
SATs: allow new records in a sequential read for full refresh test (#17660)

* SATs: allow new records in a sequential read for full refresh test

* SATs: upd changelog

* SATs: change the output when failing full refresh test

* SATs: upd according to code review

davydov-d committed:
Source facebook-marketing: remove `pixel` from custom conversions stream (#18045)

* #744 source facebook-marketing: rm pixel from custom conversions stream

* #744 source fb marketing: upd changelog

* #744 source facebook-marketing - add custom_conversions to the test catalog

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

dizel852 commented:
Reviewed the PR after fixes due to @edmundito comments

Code - LGTM.
Need to fix the last two comments(code suggestion and prop renaming)
Also tested locally(Chrome) - seems like all work as should

davydov-d committed:
#17506 fix klaviyo & marketo expected_records (#18101)

YatsukBogdan1 committed:
Update airbyte-webapp/src/App.tsx

Co-authored-by: Edmundo Ruiz Ghanem <168664+edmundito@users.noreply.github.com>

YatsukBogdan1 committed:
Updates next properties: wasActive -> isActive, lowToLarge -> isAscending

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=17798) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=17798) before we can accept your contribution.<br/>**16** out of **17** committers have signed the CLA.<br/><br/>:white_check_mark: terencecho<br/>:white_check_mark: ryankfu<br/>:white_check_mark: YatsukBogdan1<br/>:white_check_mark: evantahler<br/>:white_check_mark: xpuska513<br/>:white_check_mark: girarda<br/>:white_check_mark: sfc-gh-pkommini<br/>:white_check_mark: benmoriceau<br/>:white_check_mark: artem1205<br/>:white_check_mark: mfsiega-airbyte<br/>:white_check_mark: brianjlai<br/>:white_check_mark: Amruta-Ranade<br/>:white_check_mark: alexander-marquardt<br/>:white_check_mark: gosusnp<br/>:white_check_mark: davydov-d<br/>:white_check_mark: suhomud<br/>:x: github-actions[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=17798) it.</sub>

YatsukBogdan1 committed:
Merge branch 'master' into byatsuk/feature-create-sort-column-button
